packages/backend/src/open_payments/payment/incoming/routes.ts

```diff
@@ -73,10 +73,11 @@ async function getIncomingPaymentPublic(
   try {
     const incomingPayment = await deps.incomingPaymentService.get({
       id: ctx.params.id,
-      client: ctx.accessAction === AccessAction.Read ? ctx.client : undefined,
-      walletAddressId: ctx.walletAddress.id
+      client: ctx.accessAction === AccessAction.Read ? ctx.client : undefined
     })
-    ctx.body = incomingPayment?.toPublicOpenPaymentsType()
+    ctx.body = incomingPayment?.toPublicOpenPaymentsType(
+      deps.config.authServerGrantUrl
```

Comment:
Would there ever be a reason we'd need to return this in the private incoming payment response as well?

Reply:
I don't think so because if you can get a private one, you've already gone through the auth flow.